element-hq/element-x-android
# Display banned users in room member list

## Type of change


## Content

- Adds a segmented control button row on top of the member list when the user can ban/unban users, there are banned users and the room is not a DM.
- Display the banned users in a list in a separate 'tab'.
- When this other tab is selected, the search works on the banned users instead of the joined members.

## Motivation and context

Closes #2256.

## Screenshots / GIFs

Inside the PR.

## Tests

- Go to a room where you're not either an admin or a moderator, open the room member list. It should appear just as before.
- Now go into a room where you're either an admin or a moderator and there are banned users (EX Android one?). The segmentation control should be present.
- Change to the 'banned' tab, the banned users should be there, with no name or avatar (as they could be offensive). The click action doesn't work at the moment, it's part of a different task.
- Search for some of these users.
- Go back to the member list tab, check it works as before.

## Tested devices

- OS version(s): 14

## Checklist
